### PR TITLE
Map the WS-Fed ClaimTypes.Name to name

### PR DESCRIPTION
**Pull request by jersiovic**
It does the minimum which was requested at this issue for mapping  WS-Fed ClaimTypes.Name to name. 
https://github.com/OrchardCMS/OrchardCore/issues/9533
The part of doing it configurable is sth not in this PR 
I can try to do that job later on another PR when I end my deadline, now it is not possible for me.

### CODE REVIEW

**Comment by sebastienros** [comment]
I don't see any issue with this PR, but waiting for people with knowledge to approve.

File: src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj [hunk 1]
```diff
@@ -31,6 +31,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" />
     <PackageReference Include="OpenIddict.AspNetCore" />
+    <PackageReference Include="OpenIddict.Abstractions" />
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj:34]: The abstractions package is implicitly referenced by the `OpenIddict.AspNetCore` metapackage. Consider removing it.

File: src/OrchardCore.Build/Dependencies.props [hunk 2]
```diff
@@ -35,6 +35,7 @@
     <PackageManagement Include="Newtonsoft.Json" Version="13.0.1" />
     <PackageManagement Include="NLog.Web.AspNetCore" Version="4.12.0" />
     <PackageManagement Include="NodaTime" Version="3.0.5" />
+    <PackageManagement Include="OpenIddict.Abstractions" Version="3.0.4" />
```
Comment (kevinchalet) [inline src/OrchardCore.Build/Dependencies.props:38]: This shouldn't be necessary.

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs [hunk 3]
```diff
@@ -161,6 +163,10 @@ public class ServerStartup : StartupBase
     {
         public override void ConfigureServices(IServiceCollection services)
         {
+            services.Configure<IdentityOptions>(options =>
+            {
+                options.ClaimsIdentity.UserNameClaimType = Claims.Name;
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs:168]: It's worth noting it won't only affect access tokens but also regular ASP.NET Identity application cookies.
Instead, we may want to do the mapping in the authorization controller (see how it's done for the "sub" claim).

**Review by kevinchalet — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj (hunk 1 above)
Comment (jersiovic) [reply src/OrchardCore.Modules/OrchardCore.OpenId/OrchardCore.OpenId.csproj]: done

File: src/OrchardCore.Build/Dependencies.props (hunk 2 above)
Comment (jersiovic) [reply src/OrchardCore.Build/Dependencies.props]: removed

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs (hunk 3 above)
Comment (jersiovic) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs]: done

**Comment by jersiovic** [comment]
thank you for the review @kevinchalet

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs [hunk 4]
```diff
@@ -6,6 +6,7 @@
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Authentication.OpenIdConnect;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Identity;
```
Comment (kevinchalet) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs:9]: Are these changes necessary now that you're no longer updating the identity options?

**Comment by kevinchalet** [comment]
Looks great, thanks! Did you have a chance to confirm this fixes your issue?

**Comment by jersiovic** [comment]
yes it fixes it

File: src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs (hunk 4 above)
Comment (jersiovic) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs]: no, removed

**Review by kevinchalet — APPROVED** [review]

**Comment by kevinchalet** [comment]
Thanks for your PR!
